src/appleseed.python/bindblenderexport.cpp

```diff
@@ -84,17 +95,133 @@ namespace
-        char flag, pad;
+        int     loopstart;
+        int     totloop;
+        short   mat_nr;
+        char    flag, pad;
     };
+
+
+    //
+    // This class stores and manipulates an arbitrary number of appleseed transform sequences
+    // from Blender particle and dupli instances.  Storing them in a C++
+    // std::vector allows them to be processed and inserted into the appleseed scene
+    // without the overhead of Python.
+    //
+    
+    class BlTransformLibrary
+    {
+      public:
+        void release()
+        {
+            delete this;
+        }
+
+        std::size_t get_size() const
+        {
+            return m_xforms.size();
+        }
+
+        void add_xform_step(
+            const float         time,
+            const std::string&  instance_id,
+            const bpy::list&    matrix)
+        {
+            if (time == 0.0f)
+                m_xforms[instance_id] = TransformSequence();
+
+            // Ignore any Blender instance that doesn't already have an associated TransformSequence
+            auto instance = m_xforms.find(instance_id);
+
+            if (instance != m_xforms.end())
+            {
+                assert(bpy::len(matrix) == 16);
+
+                Matrix4d temp;
+
+                for (std::size_t index = 0; index < 16; ++index)
+                {
+                    bpy::extract<double> ex(matrix[index]);
+
+                    if (!ex.check())
+                    {
+                        PyErr_SetString(PyExc_TypeError, "Incompatible key type.");
+                        bpy::throw_error_already_set();
+                    }
+
+                    temp[index] = ex;
+                }
+
+                instance->second.set_transform(time, Transformd(temp));
+            }
+        }
+
+        void optimize_xforms()
+        {
+            for (auto& xform : m_xforms)
+                xform.second.optimize();
+        }
+
+        bool needs_assembly() const
+        {
+            assert(!m_xforms.empty());
+
+            return m_xforms.size() > 1 || m_xforms.at(0).size() > 1;
+        }
+
+        UnalignedTransformd get_single_transform() const
+        {
+            assert(m_xforms.size() == 1);
+
+            const Transformd xform(m_xforms.begin()->second.get_earliest_transform());
+            return UnalignedTransformd(xform);
+        }
+
+        void flush_instances(
+            Assembly*           as_main_ass,
+            const char*         ass_name)
+        {
+            for (auto& xform : m_xforms)
+            {
+                auto_release_ptr<AssemblyInstance> ass_inst = AssemblyInstanceFactory::create(
+                    xform.first.c_str(),
+                    ParamArray(),
+                    ass_name);
+
+                ass_inst->transform_sequence() = xform.second;
+
+                // Store pointers to assembly instances in case updates are needed.
+                m_ass_insts.emplace_back(get_pointer(ass_inst));
```

Comment:
I didn't even know we had `get_pointer()`, not sure what it's for. Typically you just do `ass_inst.get()` to get a pointer. You can leave the code as is, we'll probably remove `get_pointer()` in a separate PR.

Reply:
Ok so we'll keep `get_pointer()` (see discussion in #dev) but here it would be nicer to just use `ass_inst.get()`.